cmd/operator-sdk/run/cmd.go

```diff
@@ -114,25 +113,28 @@ https://sdk.operatorframework.io/docs/olm-integration/cli-overview
 						if err != nil {
 							return fmt.Errorf("error getting kubeconfig and default namespace: %v", err)
 						}
-						c.localArgs.watchNamespace = defaultNamespace
+						c.localArgs.WatchNamespace = defaultNamespace
 					}
 				}
 
-				c.localArgs.kubeconfig = c.kubeconfig
-				if err := c.localArgs.run(); err != nil {
+				c.localArgs.Kubeconfig = c.kubeconfig
+				if err := c.localArgs.Run(); err != nil {
 					log.Fatalf("Failed to run operator locally: %v", err)
 				}
 			}
 			return nil
 		},
 	}
-	// Avoid sorting flags so we can group them according to run type.
-	cmd.Flags().SortFlags = false
 
 	// Shared flags.
 	cmd.Flags().StringVar(&c.kubeconfig, "kubeconfig", "",
 		"The file path to kubernetes configuration file. Defaults to location "+
 			"specified by $KUBECONFIG, or to default file rules if not set")
+	err := cmd.Flags().MarkDeprecated("kubeconfig",
+		"use --kubeconfig with 'local' or 'packagemanifests' subcommands instead")
```

Comment:
Would not make more sense has a specific command to OLM features instead of re-use the run command?
Add a new subcommand as `packagemanifests` shows not be part of the scope `deprecate the --local flag in favor of a new subcommand local`